docs/en/development/io_processing.md

```diff
@@ -3,7 +3,51 @@ sidebar_label: Read/Write/Delete Processing Flow
+
+![write](../images/internals-write.png)
+
+> **NOTICE**: Compression and encryption in this figure are disabled by default. To enable these features, please add option `--compress value` or `--encrypt-rsa-key value` to the formatting command.
+
+To be more intuitive, the folloing metrics figure shows the metrics from the `stats` command.
+
+![stats](../images/internals-stats.png)
+
+From the first highlighted section in the above figure, we can see,
+
+- The average I/O size writing to the object storage is `object.put / object.put_c = 4 MiB`. It is the same as the default size of a block.
+- The number of metadata transactions to that of object storage transactions is `meta.txn : object.put_c -= 1 : 16`. It means a single slice flush requires 1 update of metadata and 16 uploads to the object storage. It also shows that every flush operation transmits 4 MiB * 16 = 64 MiB of data as the same to the default size of a chunk.
+- The average request size in the FUSE layer approximately equals to `fuse.write / fuse.ops ~= 128 KiB`, which is the same as the default limitation of the request size.
+
+It's more complicated to write a large file randomly than to write it consequently. There could be **a number of intermittent** slices in a chunk, which makes it hard to crop the data into a size of 4 MiB, and lowers the performance since it requires multiple metadata updates.
+
+Generally, a small file will be uploaded to the object storage when it is being closed, and the I/O size is the same as the file size. It could also be seen from the third highlighted section (creating 128 KiB small files) in the metrics figure above that,
+
+- The PUT size to the object storage is 128 KiB, calculated by `object.put / object.put_c`.
+- The number of metadata transactions is two times to the number of PUT in approximate, since every file requires one create and one write.
+
+It's worth to be mentioned that, when uploading objects which are smaller than a block, JuiceFS will try to write them simultaneously into the local cache (specified by `--cache-dir`, memory and disk are all available) to keep it possible to improve performance in the future. From the metrics figure above, we can see that when writing small files, the write bandwidth of the blockcache is the same to that of the object. It is meaningful that reading these small files are extremely fast since they are cached, as shown in the fourth highlighted section.
```

Comment:
```suggestion
It's worth mentioning that, when uploading objects which are smaller than a block, JuiceFS will try to write them simultaneously into the local cache (specified by `--cache-dir`, both memory and disk are available) attempting to improve performance in the future. From the metrics figure above, we can see that when writing small files, the write bandwidth of the blockcache is the same as that of the object. Since the small files are cached, reading these small files is extremely fast (as shown in the fourth highlighted section).
```

Reply:
```suggestion
It's worth mentioning that, when uploading objects which are smaller than a block, JuiceFS will try to write them simultaneously into the local cache (specified by `--cache-dir`, both memory and disk are available) attempting to improve performance in the future. From the metrics figure above, we can see that when writing small files, the write bandwidth of the blockcache is the same as that of the object. Since the small files are cached, reading these small files is extremely fast (as shown in the fourth highlighted section).
```

这里的 attempting 我不太理解，好像和原意不一样？

Reply:
用 attempting 是可以的，读起来也更流畅，我直接替换了。